### PR TITLE
requireCamelCaseOrUpperCaseIdentifiers: skip es5 getters if ignoreProper...

### LINKED ISSUE #1269: ignoreProperties for requireCamelCaseOrUpperCaseIdentifiers falsely catches getters
If I have an object like
```js
{
  get some_url() {
    return url;
  }
}
```

`some_url` is flagged as failing `requireCamelCaseOrUpperCaseIdentifiers`, even though it is only a property of the object.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/11145286-ignoreproperties-for-requirecamelcaseoruppercaseidentifiers-falsely-catches-getters?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F281640&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by zxqfox**
...ties is set

Fixes #1269

### CODE REVIEW

File: lib/rules/require-camelcase-or-uppercase-identifiers.js [hunk 1]
```diff
@@ -86,7 +86,7 @@ module.exports.prototype = {
                     return;
                 }
 
-                if (prevToken && prevToken.value === '.') {
+                if (prevToken && (prevToken.value === '.' || prevToken.type === 'Identifier')) {
```
Comment (zxqfox) [inline lib/rules/require-camelcase-or-uppercase-identifiers.js]: Agreed?

**Comment by mikesherov** [comment]
These are es5 getters.

File: lib/rules/require-camelcase-or-uppercase-identifiers.js (hunk 1 above)
Comment (mikesherov) [inline lib/rules/require-camelcase-or-uppercase-identifiers.js]: Should we explicitly check for `get` and `set` as the values of the identifiers?

Comment (zxqfox) [inline lib/rules/require-camelcase-or-uppercase-identifiers.js]: Yeah. Let's do this.

**Comment by mikesherov** [comment]
:+1: feel free to land.

**Pull request closed without merging**